### PR TITLE
Fix empty phone and email value

### PR DESCRIPTION
**Pull request by dpaun1985**
Fixes 1882-gh-Automattic/woocommerce-payments-server

#### Changes proposed in this Pull Request
Stripe does not accept empty strings for support phone and email. 
We don't have to send empty values for these fields.

#### Testing instructions

* go to `Card readers -> Receipt details` section
* set an empty value for email -> an error should be displayed and the save button disabled
* set an empty value for phone number -> an error should be displayed and the save button disabled

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_

### CODE REVIEW

File: client/card-readers/settings/sections/contacts-details.js [hunk 1]
```diff
@@ -26,12 +26,35 @@ const ContactDetailsSection = () => {
 		setAccountBusinessSupportPhone,
 	] = useAccountBusinessSupportPhone();
 
-	const businessSuppotEmailErrorMessage = useGetSavingError()?.data?.details
+	let businessSuppotEmailErrorMessage = useGetSavingError()?.data?.details
```
Comment (mgascam) [inline client/card-readers/settings/sections/contacts-details.js:29]: There appears to be a typo: `businessSupportEmailErrorMessage`

File: client/card-readers/settings/sections/contacts-details.js [hunk 2]
```diff
@@ -26,12 +26,35 @@ const ContactDetailsSection = () => {
...
 		?.account_business_support_email?.message;
 
-	const businessSuppotPhoneErrorMessage = useGetSavingError()?.data?.details
+	let businessSuppotPhoneErrorMessage = useGetSavingError()?.data?.details
```
Comment (mgascam) [inline client/card-readers/settings/sections/contacts-details.js:32]: There appears to be a typo: `businessSupportPhoneErrorMessage`

**Review by mgascam — COMMENTED** [review]
✅ During manual tests I am able to see an error when support phone or email are empty. 
✅ 'Save Changes' is disabled when support phone or email are empty
⚠️  'Save Changes' button does not enable back when a valid phone or email are filled

File: client/card-readers/settings/sections/contacts-details.js [hunk 3]
```diff
@@ -26,12 +26,35 @@ const ContactDetailsSection = () => {
...
+
+	if ( '' === accountBusinessSupportPhone ) {
+		businessSuppotPhoneErrorMessage = __(
+			'Support phone number cannot be empty!',
+			'woocommerce-payments'
+		);
+	}
+
+	const updateSaveButtonAvailability = () => {
+		if (
```
Comment (mgascam) [inline client/card-readers/settings/sections/contacts-details.js:50]: It seems that we are not accounting for the case when `accountBusinessSupportEmail` or `accountBusinessSupportPhone` are valid. I believe this prevents the 'Save changes' button to be enabled again.

File: client/card-readers/settings/sections/contacts-details.js [hunk 4]
```diff
@@ -26,12 +26,37 @@ const ContactDetailsSection = () => {
...
```
Comment (mgascam) [inline client/card-readers/settings/sections/contacts-details.js:50]: You could replace the if/else with:
`		setSaveDisabled(
			'' === accountBusinessSupportEmail ||
				'' === accountBusinessSupportPhone
		);`

**Review by mgascam — APPROVED** [review]
LGTM!

I left a couple of suggestions, feel free to skip them.

File: client/card-readers/settings/sections/contacts-details.js [hunk 5]
```diff
@@ -26,20 +26,41 @@ const ContactDetailsSection = () => {
...
+	let businessSupportEmailErrorMessage = useGetSavingError()?.data?.details
 		?.account_business_support_email?.message;
 
-	const businessSuppotPhoneErrorMessage = useGetSavingError()?.data?.details
+	let businessSupportPhoneErrorMessage = useGetSavingError()?.data?.details
 		?.account_business_support_phone?.message;
 
+	if ( '' === accountBusinessSupportEmail ) {
+		businessSupportEmailErrorMessage = __(
+			'Support email cannot be empty!',
```
Comment (kalessil) [inline client/card-readers/settings/sections/contacts-details.js:37]: Support email cannot be empty, please specify.

File: client/card-readers/settings/sections/contacts-details.js [hunk 6]
```diff
@@ -26,20 +26,41 @@ const ContactDetailsSection = () => {
...
+			'woocommerce-payments'
+		);
+	}
+
+	if ( '' === accountBusinessSupportPhone ) {
+		businessSupportPhoneErrorMessage = __(
+			'Support phone number cannot be empty!',
```
Comment (kalessil) [inline client/card-readers/settings/sections/contacts-details.js:44]: Support email cannot be empty, please specify.

File: includes/admin/class-wc-rest-payments-settings-controller.php [hunk 7]
```diff
@@ -550,8 +556,11 @@ private function update_is_wcpay_subscriptions_enabled( WP_REST_Request $request
 	 */
 	private function update_account( WP_REST_Request $request ) {
 		$updated_fields_callback = function ( $value, string $key ) {
-			return in_array( $key, static::ACCOUNT_FIELDS_TO_UPDATE, true ) &&
-					$this->wcpay_gateway->get_option( $key ) !== $value;
+			if ( ! ( in_array( $key, static::ACCOUNT_FIELDS_TO_UPDATE, true ) &&
+				$this->wcpay_gateway->get_option( $key ) !== $value ) ) {
+				return false;
+			}
+			return ! in_array( $key, static::ACCOUNT_FIELDS_CANNOT_BE_EMPTY, true ) || ! empty( $value );
```
Comment (kalessil) [inline includes/admin/class-wc-rest-payments-settings-controller.php:563]: @dpaun1985: not sure why we need changes here, would you please share insights?

**Review by kalessil — CHANGES_REQUESTED** [review]
Looks great! I left a couple of comments though.

File: includes/admin/class-wc-rest-payments-settings-controller.php (hunk 7 above)
Comment (dpaun1985) [reply includes/admin/class-wc-rest-payments-settings-controller.php:563]: > @dpaun1985: not sure why we need changes here, would you please share insights?

I wanted to add a double check of email and phone also on backend side. If the fields are empty, we don't send the empty values to Stripe. 
Another option could be also to throw an error if the fields are sent with empty values.

Comment (kalessil) [reply includes/admin/class-wc-rest-payments-settings-controller.php:563]: I see. Can we skip the changes?

There is JS validation and settings will not be saved if JS is not working - so realistically we'll not get empty values saved. Do I understand the context in the right way?

Comment (dpaun1985) [reply includes/admin/class-wc-rest-payments-settings-controller.php:563]: yes. I will revert the changes here.

Comment (kalessil) [reply includes/admin/class-wc-rest-payments-settings-controller.php:563]: Great, thanks!

File: changelog/fix-empty-phone-email-value [hunk 8]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: fix
+
+Empty values for support email and phone should not be allowed
```
Comment (kalessil) [inline changelog/fix-empty-phone-email-value:4]: -> `Prevent sending empty values for required support email and phone fields.`

**Review by mordeth — APPROVED** [review]
LGTM!

**Review by kalessil — APPROVED** [review]
LGTM!

**Pull request merged**